### PR TITLE
Resolve relative paths in exunit filter

### PR DESCRIPTION
**Pull request by sabiwara**
Tentative fix for https://github.com/elixir-lang/elixir/issues/13257

### CODE REVIEW

**Comment by josevalim** [comment]
I believe this is already fixed in main/v1.16.

**Comment by josevalim** [comment]
Actually, I am wrong, it is not. :( Maybe we need to use `Path.safe_relative`?

**Comment by sabiwara** [comment]
> Actually, I am wrong, it is not. :( 

I initially had the same thought 😄
I assumed https://github.com/elixir-lang/elixir/pull/13232 would already fix it and figured I'd just add a test for it, and noticed it wasn't.

> Maybe we need to use `Path.safe_relative`?

I used it in my original approach, but wasn't sure if we need to protect mix test against directory-traversal?
On the other hand, `mix test ../another_project/test/my_test.exs` is currently allowed, so having at least a warning might be warranted.
What do you think of [64ce95c](https://github.com/elixir-lang/elixir/pull/13258/commits/64ce95cdbedc1fb351f24dab1a2a3ceddca6949a)?

Raising is a bit hard since it would probably require a change to the signature of [`ExUnit.Filters.parse/1`](https://hexdocs.pm/ex_unit/ExUnit.Filters.html#parse/1) in order to properly raise in `Mix.Tasks.Test`?.

**Comment by josevalim** [comment]
Let’s use safe_relative and if it returns error we just keep the path as is?

**Comment by josevalim** [comment]
Ignore me, warning is good. I just realized that this new code won’t work with absolute paths neither, right?

**Comment by josevalim** [comment]
Yeah, sorry for the back-and-forth, I think the original request which used relative_to is the right to call here.

File: lib/ex_unit/lib/ex_unit/filters.ex [hunk 1]
```diff
@@ -45,7 +45,7 @@ defmodule ExUnit.Filters do
   end
 
   defp extract_line_numbers(file_path) do
-    case String.split(file_path, ":") do
+    case normalize_relative_path(file_path) |> String.split(":") do
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/filters.ex:48]:
```suggestion
    case Path.relative_to_cwd(file_path) |> String.split(":") do
```

**Comment by sabiwara** [comment]
> Yeah, sorry for the back-and-forth, I think the original request which used relative_to is the right to call here.

No worries! Reverted [64ce95c](https://github.com/elixir-lang/elixir/pull/13258/commits/64ce95cdbedc1fb351f24dab1a2a3ceddca6949a) (back to no warning).

Should I backport this after I merge?

**Review by josevalim — APPROVED** [review]
Yes, please backport!

**Pull request merged**